viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
+from ast import And
+from itertools import count
+from site import USER_BASE
+from typing import Counter
+import copy # I want to import copy module to use deepcopy
+'''I have chosen to use the deepcopy to ensure that I am not making
+changes to the original list pass through the functions.
+I am making changes to the new list for three(3) of the functions in this wave.
+However, as we know, if I only make a shallow copy, it will also change the 
+original list since the elements in both the original list and the new list
+are referencing the same ID number. By creating a deepcopy, I can ensure that as 
+the function iterates through the old list and changes to the new list,  the 
+changes will only affect the new list.'''
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    if new_movie["title"] == None or\
+        new_movie["genre"] == None or\
+            new_movie["rating"] == None:
+        return None     
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    watched_movie = copy.deepcopy(user_data)
+    watched_movie["watched"].append(movie)
+    return watched_movie

```

Comment:
So I think we accidentally put the fear of god in yall about the dangers of side effects, but sometimes we _want_ them. This is a case when we want to add to our user's data immediately without taking the time to copy it all.

Since we are simply adding a movie to the user's watched list, creating an entire copy is overkill here. Think about if we had a million movies, but only wanted to add a new one at the end. This would be very time consuming.

```suggestion
    user_data["watched"].append(movie)
    return user_data
```